Alicerce/Sources/Logging/Log+FileLogDestination.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  Log+FileLogDestination.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 07/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public extension Log {
+
+    public class FileLogDestination: LogDestination, LogDestinationFallible {
+
+        private static let dispatchQueueLabel = "com.mindera.alicerce.log.destination.file"
+
+        public var errorClosure: ((LogDestination, Item, Error) -> ())?
+
+        public let queue: Queue
+        public let minLevel: Level
+        public let formatter: LogItemFormatter
+        public private(set) var writtenItems: Int = 0
+        public var instanceId: String {
+            return "\(type(of: self))_\(fileURL.absoluteString)"
+        }
+
+        private let fileURL: URL
+        private let fileManager = FileManager.default
+
+        //MARK:- lifecycle
+
+        public init(fileURL: URL,
+                    minLevel: Level = Level.error,
+                    formatter: LogItemFormatter = StringLogItemFormatter(),
+                    queue: Queue = Queue(label: FileLogDestination.dispatchQueueLabel)) {
+
+            
+            self.fileURL = fileURL
+            self.minLevel = minLevel
+            self.formatter = formatter
+            self.queue = queue
+        }
+
+        //MARK:- public Methods
+
+        public func clear() {
+            guard fileManager.fileExists(atPath: fileURL.path) else { return }
+            do {
+                try fileManager.removeItem(at: fileURL)
+            }
+            catch {
+                print("Log file destination could not remove logfile \(fileURL).")
+            }
+        }
+
+        public func write(item: Item) {
+            queue.dispatchQueue.async { [weak self] in
+
+                guard let strongSelf = self else { return }
+
+                var reportedError: Error?
+                defer {
```

Comment:
I don't think this is a classic use of `defer`. In my opinion, we should simply call `strongSelf.errorClosure?(strongSelf, item, error)` when relevant (i.e. an error occurred), and not do this validation every time we exit the scope.